### PR TITLE
Prevent https://bugs.openjdk.org/browse/JDK-8180450 for cache code

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Review by Ladicek — APPROVED** [review]
Looks fine, though maybe @gwenneg wants to take a look as well?

Also, this indeed leads to non-idiomatic code. I believe you have benchmark numbers showing an improvement, could you share them here for justification?

**Comment by geoand** [comment]
Sure yeah. 

When I tested the change made here vs the code in `main` for the application mentioned in https://github.com/quarkusio/quarkus/issues/28795, I observed that an average request time of `40ms` was sustainable for 670k req/sec vs 650k req/sec.
That's not a very large improvement, but it's not neglibile either.

File: extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/InterceptorBindings.java [hunk 1]
```diff
@@ -13,4 +14,10 @@ public class InterceptorBindings {
     public static Set<Annotation> getInterceptorBindings(InvocationContext invocationContext) {
         return (Set<Annotation>) invocationContext.getContextData().get(ArcInvocationContext.KEY_INTERCEPTOR_BINDINGS);
     }
+
+    @SuppressWarnings("unchecked")
+    public static Set<AbstractAnnotationLiteral> getInterceptorBindingLiterals(InvocationContext invocationContext) {
```
Comment (mkouba) [inline extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/InterceptorBindings.java:19]: Could you pls add a simple comment why this method is needed?

Comment (geoand) [reply extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/InterceptorBindings.java:23]: Done

**Comment by gwenneg** [comment]
Are the changes in the cache extension temporary or they remain after the bug in the JDK has been fixed?

**Comment by geoand** [comment]
Don't hold your breath for the JDK issue to be fixed 😉 

If it ever does get fixed, then yeah, this along with a bunch of other fixes we have made can be reverted.

**Comment by franz1981** [comment]
@gwenneg sadly it won't be fixed in a way that will work with the bug that's been solved here :(
The problem happen if a concrete type perform a type check vs an interface it doesn't implement (that's different from the secondary super cache issue where the cache field ping pong between multiple implemented interfaces): it will still perform a O(N) search among all implemented interfaces , just to find that it doesn't implement it.
This won't introduce any scalability issue but will waste CPU cycles.
I am going to solve it on Netty/Vert-x side too, but reducing the Netty pipeline length by saving adding the websocket handler is still beneficial and the best solution so far.

**Pull request merged**